### ISSUE TITLE
importantForAccessibility=Yes set on parent view has no effect on iOS

### ISSUE THREAD

**Issue by qchen-ms** [microsoft/reactxp#586]
This is an iOS only issue and works fine on Android. 

I have a view component like so:
```   
<RX.View 
    importantForAccessibility={RX.Types.ImportantForAccessibility.Yes} 
    accessibilityLabel={"accessibility label"}>
  <RX.Text 
  importantForAccessibility={RX.Types.ImportantForAccessibility.NoHideDescendants} >text 1</RX.Text>
  <RX.Text 
  importantForAccessibility={RX.Types.ImportantForAccessibility.NoHideDescendants} >text 2</RX.Text>
</RX.View>
```

The expected behavior should be "accessibility label" gets read out from screen reader. However, on iOS, only "text 1" and "text 2" can be read out.

**Comment by erictraut** [comment]
@mshoho, any thoughts about this one?

**Comment by mshoho** [comment]
importantForAccessibility is an Android-only property in mobile RN (https://facebook.github.io/react-native/docs/accessibility.html#importantforaccessibility-android). And currently RX on mobile just passes the props to RN (on web we try map importantForAccessibility to aria-hidden).

I currently have web on top of my mind and will try to test this one later (it's been delayed by various things, but I'm pretty close to returning to the accessibility API refactoring). And I suppose that adding some accessibilityTrait (for example, button — depends on the actual use case) and/or touch handler to the View might help.

There is also one RN accessibility property missing in the current set of RX props — https://facebook.github.io/react-native/docs/accessibility.html#accessible-ios-android — the example there is exactly the same as the example above.

So, we can temporarily add one more platform-dependent property to current RX API, while new API is in progress. It also might make sense to reflect which accessibility properties are applicable to which platforms. But if it's not a critical issue with no workaround (which I doubt given the above info), I would rather wait for the new API.

**Comment by qchen-ms** [comment]
Hey @mshoho, 

Do you have an ETA on when the new API will be available? This isn't a pressing issue on our side yet. I haven't found a workaround using RX (tried adding trait, wasn't successful) and don't want to introduce platform specific throwaway RN code especially if a new API will come up in time :)

Thanks,
Chen

**Comment by rigdern** [comment]
I work on @mshoho's team. Let me provide more information about how we're using ReactXP's accessibility API so you can better understand its design.

In Skype, we've made some modifications to React Native's accessibility implementation. We want to contribute our changes [upstream](https://github.com/facebook/react-native) but we have to do more work first to resolve inconsistencies in the implementation between iOS and Android. We don't have an ETA for when that will be finished. Here are some details on what we've done:

In the public React Native, `importantForAccessibility` is only supported on Android. We've ported this prop to iOS. We've also ported it to Windows and the [Windows work is already publicly available](https://github.com/Microsoft/react-native-windows/blob/cd8e34add187e99f4b07cc2ca45bf52f9d45906c/ReactWindows/ReactNative.Shared/Views/View/ReactViewManager.cs#L97-L108).

We've deleted the `accessibility` prop because we've found that `importantForAccessibility` covers all scenarios covered by `accessibility` and then some.

**Comment by erictraut** [comment]
@rigdern, any update on this one? Has the PR been accepted into RN?

**Comment by rigdern** [comment]
The last time I spoke with @mshoho about this topic, the plan was to add additional iOS-only accessibility props to RN and then @mshoho was going to build a cross-platform accessibility API on top of this in RX. A new iOS-only accessibility prop was already merged into RN (accessibilityElementsHidden).

@mshoho, do you have any updates on the execution of this plan?
________________________________

**Comment by mshoho** [comment]
Newer RN versions have `accessibilityElementsHidden` property which is iOS-only way to do importantForAccessibility=NoHideDescendants.

I am still in progress with the accessibility API refactoring which is going to help addressing the platform specific madness without polluting the global API with the platform-specific accessibility props. It takes longer than expected but the current plan (if nothing will blow up again) is to start testing it internally by the end of September. And then make a public PR.

**Comment by rigdern** [comment]
@jessamp, in your example you're using the prop `accessibilityElementsHidden` rather than `importantForAccessibility`. I'm going to assume your question is about `importantForAccessibility`.

It's important to distinguish the public version of React Native from the private fork of React Native that Skype is currently using.

When building an app against the public version of React Native, `importantForAccessibility` doesn't do anything on iOS.

In Skype's private fork of React Native, we ported `importantForAccessibility` to iOS. In Skype's React Native fork, your example should behave the same way on iOS and Android.

We want to contribute a solution to this problem to Facebook's official public React Native repo. However, after thinking about the best way to do this, we decided to change our approach. Instead of adding an iOS implementation of `importantForAccessibility`, we will rely on iOS-specific React Native props like `accessible` and `accessibilityElementsHidden`. We (@mshoho) will change the implementation of ReactXP's accessibility API such that it'll consume these public iOS-only React Native accessibility props. As @mshoho mentioned in his latest post, he hopes to begin testing this solution internally at the end of September.

Does that answer your question?

**Comment by jessamp** [comment]
@rigdern yes it does. Thanks!

What do you mean by "ported `importantForAccessibility` to iOS"?

**Comment by rigdern** [comment]
@jessamp The public version of React Native has an `importantForAccessibility` prop that only works on Android. We looked at the contract that the `importantForAccessibility` props exposes to apps and then implemented this same contract on iOS.

**Comment by jessamp** [comment]
@mshoho are these changing going to utilize the new accessibility features in RN 0.57?

**Comment by fbartho** [comment]
Happy to close this in favor of #1126!
